frontend/src/store/queries/user/update-user.ts

```diff
@@ -0,0 +1,44 @@
+import { API } from '../api_routes';
+import { api } from '../index';
+
+interface UserFields {
+  sex: string;
+  birthYear: string;
+  location: string;
+  name: string;
+  phone: string;
+  email: string;
+}
+
+export interface ProfileFieldsRequestData extends UserFields {
+  password: string;
+  newPassword: string;
+  repeatNewPassword: string;
+}
+
+export interface ProfileFieldsResponseData extends UserFields {}
+
+export const updateUserApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    updateUserProfile: builder.mutation<
+      ProfileFieldsResponseData,
+      ProfileFieldsRequestData
+    >({
+      query: ({ ...put }) => ({
+        url: `${API.USER}`,
+        method: 'PUT',
+        body: put,
+      }),
```

Comment:
Why did you do the `{ ...put }`? 

const a = { foo: 'bar' };
const b = { ...a };

Both `a` and `b` have the same exact content